### PR TITLE
Dp 6246 e2e fix + datasets functionality

### PR DESCRIPTION
**Pull request by aditudorache**

### CODE REVIEW

File: modules/page/components/catalogus-themes/catalogus-themes.component.js [hunk 1]
```diff
@@ -19,8 +19,11 @@ import { routing } from '../../../../src/app/routes';
         vm.themes = CATALOGUS_THEMES_CONFIG.map(theme => {
             const linkTo = {
                 type: routing.datasets.type,
```
Comment (chrisvanmook) [inline modules/page/components/catalogus-themes/catalogus-themes.component.js]: Updating query params like this is not really . a best practice. best is to dispatch an action with an action creator. Then in the filter-query.js file we can add that action.

File: modules/shared/components/tab-header/tab-header.factory.js [hunk 2]
```diff
@@ -1,130 +1,15 @@
+import TabHeader from '../../../../src/shared/services/tab-header/tab-header';
+
 (function () {
     'use strict';
 
     angular
         .module('dpShared')
         .factory('TabHeader', tabHeaderFactory);
 
-    tabHeaderFactory.$inject = ['TAB_HEADER_CONFIG'];
```
Comment (chrisvanmook) [inline modules/shared/components/tab-header/tab-header.factory.js]: 🎉 🎉 🎉 🎉 🎉

File: src/app/components/Dataset/Dataset.jsx [hunk 3]
```diff
@@ -35,21 +37,24 @@ const Dataset = ({
...
+  apiSpecification,
+  searchText
 }) => {
   if (isLoading || (!numberOfRecords && !authError)) {
     return <LoadingIndicator />;
   }
 
   // Todo: move query and showTabbHeader logic to search page
-  const query = ''; // ?
+  const query = searchText || '';
```
Comment (chrisvanmook) [inline src/app/components/Dataset/Dataset.jsx]: could the empty sting be moved to the selector?

File: src/header/ducks/auto-suggest/auto-suggest.js [hunk 4]
```diff
@@ -41,7 +44,13 @@ export default function AutoSuggestReducer(state = initialState, action) {
...
         ...state,
         activeSuggestion: action.suggestion,
-        displayQuery: action.suggestion.label
+        displayQuery: action.suggestion.label,
+        typedQuery: action.suggestion === -1 ? '' : state.typedQuery
+      };
+    case routing.searchDatasets.type:
+      return {
+        ...state,
+        typedQuery: get(action, 'meta.query.zoekterm')
```
Comment (chrisvanmook) [inline src/header/ducks/auto-suggest/auto-suggest.js]: default value (empty string) as 3rd argument?

File: src/shared/ducks/datasets/data/data.js [hunk 5]
```diff
@@ -13,16 +15,18 @@ export const DEFAULT_VIEW = 'CATALOG';
...
-  authError: false
+  authError: false,
+  searchText: ''
 };
 
 export default function reducer(state = initialState, action) {
   switch (action.type) {
     case routing.datasets.type: {
-      const { page } = action.meta.query || {};
+      const { page, zoekterm } = action.meta.query || {};
```
Comment (chrisvanmook) [inline src/shared/ducks/datasets/data/data.js]: I'm not 100% sure, but we have a dataSearch and I might think that is the place where you want to save the searchText in the store

File: src/shared/services/tab-header/tab-header.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                  optie 'Alle datasets tonen' en filter vervolgens op thema.`
+    }
+  }
+};
+
+const countProvider = {};
+const tabHeader = {};
+
+export class TabPage {
+  constructor(id, { title, action, getPayload, tip }) {
```
Comment (chrisvanmook) [inline src/shared/services/tab-header/tab-header.js]: awwww :( for a second thought this TabHeader was removed for ever ;)

File: modules/page/components/catalogus-themes/catalogus-themes.component.js (hunk 1 above)
Comment (aditudorache) [reply modules/page/components/catalogus-themes/catalogus-themes.component.js]: action creator moved to redux-first-router

File: modules/shared/components/tab-header/tab-header.factory.js (hunk 2 above)
Comment (aditudorache) [reply modules/shared/components/tab-header/tab-header.factory.js]: :-)

File: src/app/components/Dataset/Dataset.jsx (hunk 3 above)
Comment (aditudorache) [reply src/app/components/Dataset/Dataset.jsx]: done

File: src/header/ducks/auto-suggest/auto-suggest.js (hunk 4 above)
Comment (aditudorache) [reply src/header/ducks/auto-suggest/auto-suggest.js]: thanks!

File: src/shared/services/api/api.js [hunk 7]
```diff
@@ -1,19 +1,23 @@
...
 import SHARED_CONFIG from '../shared-config/shared-config';
 import { encodeQueryParams } from '../query-string-parser/query-string-parser';
+import { logout } from '../auth/auth';
 
 export const getAccessToken = () => getState().user.accessToken;
 
 export const generateParams = (data) => Object.entries(data).map((pair) => pair.map(encodeURIComponent).join('=')).join('&');
 
 const handleErrors = (response) => {
+  // Todo:
```
Comment (aditudorache) [inline src/shared/services/api/api.js]: Ticket?

File: src/shared/services/api/api.js [hunk 8]
```diff
@@ -1,19 +1,23 @@
...
+    logout();
+  }
   if (!response.ok) {
     throw Error(response.statusText);
   }
   return response;
 };
 
-
 // TODO: This function is not used yet because it is not finished
```
Comment (aditudorache) [inline src/shared/services/api/api.js]: TIcket

**Pull request merged**